war/src/main/js/add-item.js

```diff
@@ -0,0 +1,414 @@
+// Initialize all modules by requiring them. Also makes sure they get bundled (see gulpfile.js).
+var $jq = require('jquery-detached').getJQuery();
+var jenkinsLocalStorage = require('./util/jenkinsLocalStorage.js');
+var itemIcons = {
+    'hudson.model.FreeStyleProject':'freestyle-48.png',
+    'hudson.maven.MavenModuleSet':'maven-48.png',
+    'org.jenkinsci.plugins.workflow.job.WorkflowJob':'pipeline-48.png',
+    'com.infradna.hudson.plugins.backup.BackupProject':'backup-48.png',
+    'hudson.model.ExternalJob':'remote-job-48.png',
+    'com.cloudbees.jenkins.plugins.longrunning.LongRunningProject':'long-running-48.png',
+    'hudson.matrix.MatrixProject':'multi-config-48.png',
+    'com.cloudbees.hudson.plugins.folder.Folder':'folder-48.png',
+    'org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject':'branch-project-48.png',
+    'jenkins.branch.OrganizationFolder.org.jenkinsci.plugins.github_branch_source.GitHubSCMNavigator':'gitHub-project-48.png',
+    'jenkins.branch.OrganizationFolder.com.cloudbees.jenkins.plugins.bitbucket.BitbucketSCMNavigator':'bitBucket-project-48.png',
+    'copy':'copy-48.png',
+    'com.cloudbees.hudson.plugins.modeling.impl.jobTemplate.JobTemplate':'job-template-48.png',
+    'com.cloudbees.hudson.plugins.modeling.impl.folder.FolderTemplate':'folder-template-48.png',
+    'com.cloudbees.hudson.plugins.modeling.impl.auxiliary.AuxModel':'aux-template-48.png', 
+    'com.cloudbees.hudson.plugins.modeling.impl.builder.BuilderTemplate':'builder-template-48.png',
+    'com.cloudbees.hudson.plugins.modeling.impl.publisher.PublisherTemplate':'publish-template-48.png'
+} 
+    
+    
+    
+   
+var itemTypes = [
+    {
+      order : 0,
+      id : 'basic',
+      display : 'Basic items',
+      description : 'Traditional Jenkins items, typically job like entities for Jenkins to process in accordance with its configuration settings.',
```

Comment:
So this is the major part of the complete fiction that is going on here.
Manuel is already tasked with figuring out how to move this logic to the server. There is real both technical and potentially philosophical debate with how these categories work. But that debate really has nothing to do with the UI end product, so I need a way to side-step that discussion.

...so that is what this hack-pile is all about.